l1-contracts/src/core/Rollup.sol

```diff
@@ -263,55 +277,67 @@ contract Rollup is EIP712("Aztec Rollup", "1"), Leonidas, IRollup, ITestRollup {
    * @dev     We provide the `_archive` and `_blockHash` even if it could be read from storage itself because it allow for
    *          better error messages. Without passing it, we would just have a proof verification failure.
    *
-   * @param  _epochSize - The size of the epoch (to be promoted to a constant)
-   * @param  _args - Array of public inputs to the proof (previousArchive, endArchive, previousBlockHash, endBlockHash, endTimestamp, outHash, proverId)
-   * @param  _fees - Array of recipient-value pairs with fees to be distributed for the epoch
-   * @param  _aggregationObject - The aggregation object for the proof
-   * @param  _proof - The proof to verify
+   * @param _args - The arguments to submit the epoch root proof:
+   *          _epochSize - The size of the epoch (to be promoted to a constant)
+   *          _args - Array of public inputs to the proof (previousArchive, endArchive, previousBlockHash, endBlockHash, endTimestamp, outHash, proverId)
+   *          _fees - Array of recipient-value pairs with fees to be distributed for the epoch
+   *          _aggregationObject - The aggregation object for the proof
+   *          _proof - The proof to verify
    */
-  function submitEpochRootProof(
-    uint256 _epochSize,
-    bytes32[7] calldata _args,
-    bytes32[] calldata _fees,
-    bytes calldata _aggregationObject,
-    bytes calldata _proof
-  ) external override(IRollup) {
+  function submitEpochRootProof(SubmitEpochRootProofArgs calldata _args) external override(IRollup) {
     if (canPrune()) {
       _prune();
     }
 
-    uint256 previousBlockNumber = tips.provenBlockNumber;
-    uint256 endBlockNumber = previousBlockNumber + _epochSize;
+    SubmitEpochRootProofInterimValues memory interimValues;
+
+    interimValues.previousBlockNumber = tips.provenBlockNumber;
+    interimValues.endBlockNumber = interimValues.previousBlockNumber + _args.epochSize;
 
     // @note The getEpochForBlock is expected to revert if the block is beyond pending.
     //       If this changes you are gonna get so rekt you won't believe it.
     //       I mean proving blocks that have been pruned rekt.
-    Epoch epochToProve = getEpochForBlock(endBlockNumber);
+    interimValues.epochToProve = getEpochForBlock(interimValues.endBlockNumber);
+    interimValues.startEpoch = getEpochForBlock(interimValues.previousBlockNumber + 1);
+
+    // Ensure that the proof is not across epochs
+    require(
+      interimValues.startEpoch == interimValues.epochToProve,
+      Errors.Rollup__InvalidEpoch(interimValues.startEpoch, interimValues.epochToProve)
+    );
```

Comment:
interimValues.epochToProve must equal proofClaim.epochToProve, i may be missing where this assertion is

Reply:
Ok the assertion is at the bottom, they dont get their bond back if they do not prove the epoch in the claim

Reply:
We don't have that assertion, it is an if for freeing bond. Will move it up to make it easier to follow 🫡